pkg/asset/installconfig/ovirt/credentials.go

```diff
@@ -1,92 +1,272 @@
 package ovirt
 
 import (
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
 	"fmt"
-	"net/url"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
 
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	"gopkg.in/AlecAivazis/survey.v1"
 )
 
-func askCredentials() (Config, error) {
-	c := Config{}
+// clientHTTP struct - Hold info about http calls
+type clientHTTP struct {
+	saveFilePath string // Path for saving file (GET method)
+	urlAddr      string // URL or Address
+	skipVerify   bool   // skipt cert validatin in the http call
+	certPool     *x509.CertPool
+}
+
+// EngineConfig struct - Hold all info about user environment
+var EngineConfig = Config{}
```

Comment:
I would recommend not creating global shared state.

Secondly, Why do we need to provide this as part of public api of the package?